src/Cli/Utils.cs

```diff
@@ -904,27 +904,37 @@ public static bool WriteJsonContentToFile(string file, string jsonContent)
         /// This method will check if DAB_ENVIRONMENT value is set.
         /// If yes, it will try to merge dab-config.json with dab-config.{DAB_ENVIRONMENT}.json
         /// </summary>
-        public static string MergeConfigsIfAvailable()
+        /// <returns>Returns the name of the merged Config if successful.</returns>
+        public static bool TryMergeConfigsIfAvailable(out string mergedFile)
         {
             string? environmentValue = Environment.GetEnvironmentVariable(RUNTIME_ENVIRONMENT_VAR_NAME);
+            mergedFile = string.Empty;
             if (!string.IsNullOrEmpty(environmentValue))
             {
-                string FileToBeOverridden = DefaultName;
-                string FileToBeOverriddenWith = $"{CONFIGFILE_NAME}.{environmentValue}{CONFIG_EXTENSION}";
-                string mergedFile = $"{CONFIGFILE_NAME}.{environmentValue}.merged{CONFIG_EXTENSION}";
+                string baseConfigFile = DefaultName;
+                string overrideConfigFile = RuntimeConfigPath.GetFileName(environmentValue, considerOverrides: false);
+                mergedFile = $"{CONFIGFILE_NAME}.{environmentValue}.merged{CONFIG_EXTENSION}";
 
-                if (DoesFileExistInCurrentDirectory(FileToBeOverridden) && DoesFileExistInCurrentDirectory(FileToBeOverriddenWith))
+                if (DoesFileExistInCurrentDirectory(baseConfigFile) && !string.IsNullOrEmpty(overrideConfigFile))
```

Comment:
Do you think there may be a name besides `overrideConfigFile` which would be explicitly clear to distinguish:
- file which will be merged into the base file
- file with 'overrides' in the filename, which per L915 `RuntimeConfigPath.GetFileName(environmentValue, considerOverrides: false);` is not considering overrides files.

The usage of the term 'overrides' is now ambiguous.

Reply:
updated the variable name to `environmentBasedConfigFile` as file name is `dab-config.{DAB_ENVIRONMENT}.json`, so this seems more correct considering it will have changes specific to environment.